### ISSUE TITLE
docker_container - Volume With Root Host Directory Fails (cAdvisor)

### ISSUE THREAD

**Issue by amcrn** [ansible/ansible-modules-core#3823]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

docker_container
##### ANSIBLE VERSION

```
ansible 2.1.0.0 (detached HEAD b599477242) last updated 2016/05/31 22:32:16 (GMT +000)
  lib/ansible/modules/core: (detached HEAD 04a871d007) last updated 2016/05/31 22:32:28 (GMT +000)
  lib/ansible/modules/extras: (detached HEAD b5fa2b3416) last updated 2016/05/31 22:32:34 (GMT +000)
  config file = /ansible/ansible.cfg
  configured module search path = ['/opt/ansible/ansible/library']
```
##### CONFIGURATION

None.
##### OS / ENVIRONMENT

N/A
##### SUMMARY

If a docker_container has a volume with the root directory of the host, it fails with `"msg": "Error creating container: 500 Server Error: Internal Server Error (\"Invalid volume spec \"/\": Invalid specification: destination can't be '/' in '/'\")"`
##### STEPS TO REPRODUCE

```

---
- hosts: THE_HOST
  tasks:
  - name: start cadvisor container
    docker_container:
      name: cadvisor-test
      image: "google/cadvisor:v0.23.2"
      state: "started"
      ports:
      - "8080:8080"
      network_mode: host
      detach: True
      volumes:
      - "/:/rootfs:ro"
      - "/var/run:/var/run:rw"
      - "/sys:/sys:ro"
      - "/var/lib/docker/:/var/lib/docker:ro"
```

##### EXPECTED RESULTS

Container starts successfully.
##### ACTUAL RESULTS

```
TASK [start cadvisor container] ************************************************
...
fatal: [10.65.202.118]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_args": {"api_version": null, "blkio_weight": null, "cacert_path": null, "capabilities": null, "cert_path": null, "command": null, "cpu_period": null, "cpu_quota": null, "cpu_shares": null, "cpuset_cpus": null, "cpuset_mems": null, "debug": false, "detach": true, "devices": null, "dns_opts": null, "dns_search_domains": null, "dns_servers": null, "docker_host": null, "entrypoint": null, "env": null, "env_file": null, "etc_hosts": null, "exposed_ports": null, "filter_logger": false, "force_kill": false, "groups": null, "hostname": null, "image": "google/cadvisor:v0.23.2", "interactive": false, "ipc_mode": null, "keep_volumes": true, "kernel_memory": null, "key_path": null, "kill_signal": null, "labels": null, "links": null, "log_driver": "json-file", "log_options": null, "mac_address": null, "memory": "0", "memory_reservation": null, "memory_swap": null, "memory_swappiness": null, "name": "cadvisor-test", "network_mode": "host", "networks": null, "oom_killer": null, "paused": false, "pid_mode": null, "ports": ["8080:8080"], "privileged": false, "published_ports": ["8080:8080"], "pull": false, "purge_networks": null, "read_only": false, "recreate": false, "restart": false, "restart_policy": null, "restart_retries": 0, "security_opts": null, "shm_size": null, "ssl_version": null, "state": "started", "stop_signal": null, "stop_timeout": null, "timeout": null, "tls": null, "tls_hostname": null, "tls_verify": null, "trust_image_content": false, "tty": false, "ulimits": null, "user": null, "uts": null, "volume_driver": null, "volumes": ["/:/rootfs:ro", "/var/run:/var/run:rw", "/sys:/sys:ro", "/var/lib/docker/:/var/lib/docker:ro"], "volumes_from": null}, "module_name": "docker_container"}, "msg": "Error creating container: 500 Server Error: Internal Server Error (\"Invalid volume spec \"/\": Invalid specification: destination can't be '/' in '/'\")"}
```
##### NOTES

The equivalent `docker run` command works as expected:

```
docker run \
  --volume=/:/rootfs:ro \
  --volume=/var/run:/var/run:rw \
  --volume=/sys:/sys:ro \
  --volume=/var/lib/docker/:/var/lib/docker:ro \
  --publish=8080:8080 \
  --detach=true \
  --name=cadvisor-test \
  google/cadvisor:v0.23.2
```

This issue does not exist in the (deprecated) `docker` module, only the `docker_container` module.

**Comment by chouseknecht** [comment]
This is fixed by commit: e2d8d9d09a0a62d7aaa2ac915ef56d5a10fc673e. I tested with the following: 

```
- docker_container:
    name: sleepy
    image: ubuntu:14.04
    command: sleep infinity
    volumes:
      - "/:/rootfs:ro"
    register: output
```

In the resulting container it creates the correct binds:

```
"Binds": [
    "/:/rootfs:ro"
],
```

And the correct volumes:

```
"Volumes": {
    "/rootfs": {}
 },
```

**Comment by amcrn** [comment]
Appreciate the fix @chouseknecht , thanks!
